integrationTests/main/src/test/java/com/grafana/opentelemetry/IntegrationTest.java

```diff
@@ -39,59 +40,54 @@ class IntegrationTest {
 
   @Autowired private GrafanaProperties properties;
 
+  @SuppressWarnings("unused")
+  @Autowired
+  private ConnectionProperties connectionProperties;
+
   @Autowired private Optional<AutoConfiguredOpenTelemetrySdk> sdk;
 
-  static {
-    String delay = "500";
-    System.setProperty("otel.metric.export.interval", delay);
-    System.setProperty("otel.bsp.schedule.delay", delay);
-    System.setProperty("otel.exporter.otlp.protocol", "http/protobuf");
+  @Test
+  void testProperties() {
+    assertThat(properties.getCloud().getZone()).isEqualTo("prod-eu-west-0");
   }
 
   @Test
-  void testProperties() {
-    Assertions.assertThat(properties.getCloud().getZone()).isEqualTo("prod-eu-west-0");
+  void metricsResourceAttributes() throws IllegalAccessException {
+    List<Map.Entry<String, String>> metricsResourceAttributes = getMetricsResourceAttributes();
+    assertThat(metricsResourceAttributes)
+        .contains(Map.entry("telemetry.sdk.name", "io.micrometer"));
+    assertThat(metricsResourceAttributes).extracting(Map.Entry::getKey).doesNotHaveDuplicates();
   }
 
   @Test
-  void systemPropHasPriority() {
-    Assertions.assertThat(sdk)
+  void usesHttpByDefault() {
+    assertThat(sdk)
         .hasValueSatisfying(
             v -> {
               try {
                 ConfigProperties p =
                     (ConfigProperties) MethodUtils.invokeMethod(v, true, "getConfig");
-                Assertions.assertThat(p.getString("otel.exporter.otlp.protocol"))
-                    .isEqualTo("http/protobuf");
+                assertThat(p.getString("otel.exporter.otlp.protocol")).isEqualTo("http/protobuf");
               } catch (Exception e) {
                 throw new RuntimeException(e);
               }
             });
   }
 
-  @Test
-  void dataIsSent() {
-    restTemplate.getForEntity("/hello", String.class);
-
-    await()
-        .atMost(10, SECONDS)
-        .untilAsserted(
-            () -> {
-              verifyPath("/v1/traces");
-              verifyPath("/v1/metrics");
-              verifyPath("/v1/logs");
-            });
-  }
+  private static List<Map.Entry<String, String>> getMetricsResourceAttributes()
+      throws IllegalAccessException {
+    CompositeMeterRegistry globalRegistry = Metrics.globalRegistry;
```

Comment:
Would it make sense using an `OtlpMeterRegistry ` instance that is injected instead of the global so that you can avoid reflection (also, static things can make tests brittle).

Reply:
good idea!